### PR TITLE
optilize merkle for tps

### PR DESCRIPTION
**Pull request by qwdarrenyin**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=4577109&bid=15378904)

File: libdevcore/TrieHash2.cpp [hunk 1]
```diff
@@ -33,255 +35,87 @@
...
+    if (_bytesCaches.empty())
     {
-        unsigned x = std::min(
-            sharedPre, std::min((unsigned)_begin->first.size(), (unsigned)i->first.size()));
-        unsigned shared = _preLen;
-        for (; shared < x && _begin->first[shared] == i->first[shared]; ++shared)
-        {
-        }
-        sharedPre = std::min(shared, sharedPre);
+        return dev::sha3(bytes());
```
Comment (bxq2011hust) [inline libdevcore/TrieHash2.cpp]: may be you can use Invalid256 ?

File: libethcore/Block.cpp [hunk 2]
```diff
@@ -310,6 +325,7 @@ void Block::calReceiptRoot(bool update) const
 {
     if (g_BCOSConfig.version() >= V2_2_0)
     {
+        // calReceiptRootV2_2_0(update);
```
Comment (bxq2011hust) [inline libethcore/Block.cpp]: same as below?

File: libethcore/Block.cpp [hunk 3]
```diff
@@ -149,6 +149,7 @@ void Block::calTransactionRoot(bool update) const
 {
     if (g_BCOSConfig.version() >= V2_2_0)
     {
+        // calTransactionRootV2_2_0(update);
```
Comment (bxq2011hust) [inline libethcore/Block.cpp]: same as below

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=4577109&bid=15378904)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=4577109&bid=15378904)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=4577109&bid=15378904)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=4577109&bid=15378904)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=4577109&bid=15378904)

**Pull request merged**